earth2studio/models/px/dlesym.py

```diff
@@ -0,0 +1,1077 @@
+    ----
+    *args
+        Arguments for :class:`DLESyM`
+    **kwargs
+        Keyword arguments for :class:`DLESyM`
+    """
+
+    @check_extra_imports("dlesym", [OmegaConf, Module, insolation, earth2grid])
+    def __init__(
+        self,
+        atmos_model: torch.nn.Module,
+        ocean_model: torch.nn.Module,
+        hpx_lat: np.ndarray,
+        hpx_lon: np.ndarray,
+        nside: int,
+        center: np.ndarray,
+        scale: np.ndarray,
+        atmos_constants: np.ndarray,
+        ocean_constants: np.ndarray,
+        atmos_input_times: np.ndarray,
+        ocean_input_times: np.ndarray,
+        atmos_output_times: np.ndarray,
+        ocean_output_times: np.ndarray,
+        atmos_variables: list[str],
+        ocean_variables: list[str],
+        atmos_coupling_variables: list[str],
+        ocean_coupling_variables: list[str],
+    ):
+        super().__init__(
+            atmos_model=atmos_model,
+            ocean_model=ocean_model,
+            hpx_lat=hpx_lat,
+            hpx_lon=hpx_lon,
+            nside=nside,
+            center=center,
+            scale=scale,
+            atmos_constants=atmos_constants,
+            ocean_constants=ocean_constants,
+            atmos_input_times=atmos_input_times,
+            ocean_input_times=ocean_input_times,
+            atmos_output_times=atmos_output_times,
+            ocean_output_times=ocean_output_times,
+            atmos_variables=atmos_variables,
+            ocean_variables=ocean_variables,
+            atmos_coupling_variables=atmos_coupling_variables,
+            ocean_coupling_variables=ocean_coupling_variables,
+        )
+
+        self.hpx_grid = earth2grid.healpix.Grid(
+            level=int(np.log2(self.nside)),
+            pixel_order=earth2grid.healpix.HEALPIX_PAD_XY,
+        )
+        self.ll_grid = earth2grid.latlon.equiangular_lat_lon_grid(721, 1440)
+        self.regrid_to_hpx = earth2grid.get_regridder(self.ll_grid, self.hpx_grid).to(
+            torch.float32
+        )
+        self.regrid_to_ll = earth2grid.get_regridder(self.hpx_grid, self.ll_grid).to(
+            torch.float32
+        )
+        self.lat = np.linspace(90, -90, 721, endpoint=True)
+        self.lon = np.linspace(0, 360, 1440, endpoint=False)
+
+    def input_coords(self) -> CoordSystem:
+        """Input coordinate system of prognostic model
+
+        Returns
+        -------
+        CoordSystem
+            Coordinate system dictionary
+        """
+        coords = super().input_coords()
+        coords = self.coords_to_ll(coords)
+        return coords
+
+    def output_coords(self, input_coords: CoordSystem) -> CoordSystem:
+        """Output coordinate system of the prognostic model
+
+        Parameters
+        ----------
+        input_coords : CoordSystem
+            Input coordinate system
+
+        Returns
+        -------
+        CoordSystem
+            Output coordinate system
+        """
+        coords = super().output_coords(input_coords)
```

Comment:
same, comment to add some input coord validation, also input coords will be lat/lon. Will this work calling super output corrds?